### PR TITLE
docs: add node configuration docs.

### LINKED ISSUE #1849: Document configuration options
We do have some documentation for configuration options in Go code, but that's not visible to general users, they look in config files, see some fifty random options there and they have no idea how to use at least half of them, so we need better documentation.

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Close #1849.

### CODE REVIEW

File: docs/cli.md [hunk 1]
```diff
@@ -35,6 +35,229 @@ If you want to use some non-default configuration directory path, specify
 
 The file loaded is chosen automatically depending on network mode flag.
 
+Each config file contains two sections. `ApplicationConfiguration` describes node-related
```
Comment (AnnaShaleva) [inline docs/cli.md:38]: But I'm not sure that `cli` is a proper place for this.

File: docs/cli.md [hunk 2]
```diff
@@ -35,6 +35,229 @@ If you want to use some non-default configuration directory path, specify
...
+node-related settings described in the table below.
+
+| Section | Type | Example value | Description |
+| --- | --- | --- | --- |
+| Address | `string` | `127.0.0.1` | Node address. |
+| AnnouncedPort | `uint16` | `20333` | Node port which should be used for P2P version exchange. It may differs from `NodePort` node is bound to. |
+| AttemptConnPeers | `int` | `10` |  Number of connection to try to establish when the connection count drops below the `MinPeers` value.|
+| DBConfiguration | [DB Configuration](#DB-Configuration) |  | Describes configuration for database. See the [DB Configuration](#DB-Configuration) section for details. |
+| DialTimeout | `int64` | `3` | Maximum duration a single dial may take in seconds. |
+| ExtensiblePoolSize | `int` | `20` | Maximum amount of the extensible payloads from a single sender. |
```
Comment (roman-khimov) [inline docs/cli.md:56]: ... stored in a local pool

File: docs/cli.md [hunk 3]
```diff
@@ -35,6 +35,229 @@ If you want to use some non-default configuration directory path, specify
...
+| LogPath | `string` | `./logs` | Path where to store node logs. |
```
Comment (roman-khimov) [inline docs/cli.md:57]: File path?

File: docs/cli.md [hunk 4]
```diff
@@ -35,6 +35,229 @@ If you want to use some non-default configuration directory path, specify
...
+| Prometheus | [Metrics Services Configuration](#Metrics-Services-Configuration) | | Configuration for Prometheus (monitoring system). See the [Metrics Services Configuration](#Metrics-Services-Configuration) section for details |
+| ProtoTickInterval | `int64` | `5` | Duration in seconds between protocol ticks with each connected peer. |
+| Relay | `bool` | `true` | Determines whether the server is forwarding its inventory. |
+| RPC | [RPC Configuration](#RPC-Configuration) |  | Describes [RPC subsystem](rpc.md) configuration. See the [RPC Configuration](#RPC-Configuration) for details. |
+| StateRoot | [State Root Configuration](#State-Root-Configuration) |  | State root module configuration. See the [State Root Configuration](#State-Root-Configuration) section for details. |
+| UnlockWallet | [Unlock Wallet Configuration](#Unlock-Wallet-Configuration) |  | Node wallet configuration. See the [Unlock Wallet Configuration](#Unlock-Wallet-Configuration) section for details. |
```
Comment (roman-khimov) [inline docs/cli.md:71]: used for consensus (dBFT) operation

File: docs/cli.md [hunk 5]
```diff
@@ -35,6 +35,229 @@ If you want to use some non-default configuration directory path, specify
...
+  RefreshInterval: 180s
+  RequestTimeout: 5s
+  ResponseTimeout: 5s
+  UnlockWallet:
+    Path: "./oracle_wallet.json"
+    Password: "pass"
+```
+where:
+- `Enabled` marks whether Oracle module is enabled.
+- `AllowPrivateHost` denotes whether requested URI doesn't need to be validated.
```
Comment (roman-khimov) [inline docs/cli.md:124]: for private IP ranges

File: docs/cli.md [hunk 6]
```diff
@@ -35,6 +35,229 @@ If you want to use some non-default configuration directory path, specify
...
+    Address: ""
+    CertFile: serv.crt
+    Enabled: true
+    Port: 10331
+    KeyFile: serv.key
+```
+where:
+- `Enabled` denotes whether RPC server should be started.
+- `Address` is an RPC server address to be running at.
+- `EnableCORSWorkaround` enables Cross-Origin Resource Sharing.
```
Comment (roman-khimov) [inline docs/cli.md:197]: and is useful if you're accessing RPC interface from the browser

File: docs/cli.md [hunk 7]
```diff
@@ -35,6 +35,229 @@ If you want to use some non-default configuration directory path, specify
...
+where:
+- `Path` is a path to wallet.
+- `Password` is a wallet password.
+
+#### Protocol configuration
+
+ProtocolConfiguration section of `yaml` node configuration file contains
+protocol-related settings described in the table below.
+
+| Section | Type | Example value | Description | Notes |
```
Comment (roman-khimov) [inline docs/cli.md:238]: Default value?

File: docs/cli.md [hunk 8]
```diff
@@ -35,6 +35,229 @@ If you want to use some non-default configuration directory path, specify
...
+| Magic | `uint32` | `5195086` | Magic number which uniquely identifies NEO network. |
+| MaxBlockSize | `uint32` | `16777216` | Maximum block size in bytes. |
+| MaxBlockSystemFee | `int64` | `900000000000` | Maximum overall transactions system fee per block. |
+| MaxTraceableBlocks | `uint32` | `2102400` |  Length of the chain accessible to smart contracts. | `RemoveUntraceableBlocks` should be enabled to use this setting. |
+| MaxTransactionsPerBlock | `uint16` | `512` | Maximum number of transactions per block. |
+| MemPoolSize | `int` | `50000` | Size of the node's memory pool where transactions are stored before they are added to block. |
+| NativeActivations | `map[string][]uint32` | ContractManagement: [0]<br>StdLib: [0]<br>CryptoLib: [0]<br>LedgerContract: [0]<br>NeoToken: [0]<br>GasToken: [0]<br>PolicyContract: [0]<br>RoleManagement: [0]<br>OracleContract: [0] | The list of histories of native contracts updates. Each list item shod be presented as a known native contract name with the corresponding list of chain's heights. The contract is not active until chain reaches the first height value specified in the list. | `Notary` native activation is supported. |
```
Comment (roman-khimov) [inline docs/cli.md:247]: Notary is supported?

File: docs/cli.md [hunk 9]
```diff
@@ -35,6 +35,229 @@ If you want to use some non-default configuration directory path, specify
...
+| P2PNotaryRequestPayloadPoolSize | `int` | `100` | Size of the node's P2P Notary request payloads memory pool where P2P Notary requests are stored before main or fallback transaction is completed and added to the chain.<br>This option is valid only if `P2PSigExtensions` are enabled. | Not supported by the C# node, thus may affect heterogeneous networks functionality. |
+| P2PSigExtensions | `bool` | `false` | Enables following additional Notary service related logic:<br>• Transaction attributes `NotValidBefore`, `Conflicts` and `NotaryAssisted`<br>• Network payload of the `P2PNotaryRequest` type<br>• Native `Notary` contract<br>• Notary node module | Not supported by the C# node, thus may affect heterogeneous networks functionality. |
+| RemoveUntraceableBlocks | `bool`| `false` | Denotes whether old blocks should be removed from cache and database. If enabled, then only last `MaxTraceableBlocks` are stored and accessible to smart contracts. |
+| ReservedAttributes | `bool` | `false` | Allows to have reserved attributes range for experimental or private purposes. |
+| SaveStorageBatch | `bool` | `false` | Enables storage batch saving before every persist. |
```
Comment (roman-khimov) [inline docs/cli.md:252]: similar to StorageDump plugin for C# node?

File: docs/cli.md [hunk 10]
```diff
@@ -35,6 +35,229 @@ If you want to use some non-default configuration directory path, specify
...
+| SecondsPerBlock | `int` | `5` | Minimal time that should pass before next block is accepted. |
+| SeedList | `[]string` | ["seed1.neo.org:10333", "seed2.neo.org:10333", "seed3.neo.org:10333"] | List of initial nodes addresses used to establish connectivity. |
+| StandbyCommittee | `[]string` | ["03b209fd4f53a7170ea4444e0cb0a6bb6a53c2bd016926989cf85f9b0fba17a70c", "02df48f60e8f3e01c48ff40b9b7f1310d7a8b2a193188befe1c2e3df740e895093"] | List of public keys of standby committee validators are chosen from. |
+| StateRootInHeader | `bool` | `true` | Enables storing state root in block header. |
```
Comment (roman-khimov) [inline docs/cli.md:256]: Experimental protocol extension!

File: docs/cli.md (hunk 7 above)
Comment (AnnaShaleva) [reply docs/cli.md]: Replaced by defaults.

File: docs/cli.md (hunk 8 above)
Comment (AnnaShaleva) [reply docs/cli.md]: Yes. Fixed the text.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**